rasa/engine/graph.py

```diff
@@ -149,9 +178,27 @@ def _load_component(
             f"{self._component_class.__name__}.{self._constructor_name} "
             f"with config: {self._component_config}, and kwargs: {kwargs}."
         )
+
         self._component: GraphComponent = getattr(  # type: ignore[no-redef]
             self._component_class, self._constructor_name
-        )(self._component_config, self._execution_context, **kwargs)
+        )(
+            self._component_config,
+            self._model_storage,
+            self._get_resource(kwargs),
+            execution_context=self._execution_context,
+            **kwargs,
+        )
+
+    def _get_resource(self, kwargs: Dict[Text, Any]) -> Resource:
+        if "resource" in kwargs:
```

Comment:
Do we need to make this more robust?

Reply:
Yeah, I thought of looking at the type, too 👍🏻 Just the type seems risky in case multiple resources of type `Resource` are passed in. I'd go for explicit now and maybe open it up later?

Reply:
actually the incompatible type should be caught in the graph validation, no?

Reply:
incompatible types yes, but not if you had some other, compatible type called `resource`.

Reply:
Can we also validate that they don't return incompatible types to `resource`? The param `resource` is reserved by us. They shouldn't use it for anything else. Otherwise this breaks our logic of passing in `resource`s (e.g. how do we pass the resource if they are already passing something else for this param?)

Reply:
Yes good idea. Let's check for all reserved argument names.

Reply:
I take a note in the validation issue